pkg/reconciler/knativeserving/knativeserving_controller.go

```diff
@@ -233,6 +225,63 @@ func (r *ReconcileKnativeServing) install(instance *servingv1alpha1.KnativeServi
 	return nil
 }
 
+func (r *ReconcileKnativeServing) executeInstall(extensions common.Extensions, instance *servingv1alpha1.KnativeServing) error {
+	err := r.config.Transform(extensions.Transform(r.scheme, instance)...)
+	if err != nil {
+		return err
+	}
+	err = extensions.PreInstall(instance)
+	if err != nil {
+		return err
+	}
+	err = r.applyEnabledResources(instance)
+	if err != nil {
+		return err
+	}
+	return extensions.PostInstall(instance)
+}
+
+func (r *ReconcileKnativeServing) applyEnabledResources(instance *servingv1alpha1.KnativeServing) error {
+	for _, spec := range r.config.Resources {
+		if !shouldApplySpec(spec, instance) {
+			continue
```

Comment:
I think instead of `continue` we should put `r.config.Delete(&spec)`. I believe that will account for updates to the CR that add/remove disabled/skipped components.

Reply:
Thanks, good suggestion